sample-xml/app/src/main/java/co/nimblehq/sample/xml/ui/screens/home/HomeViewModel.kt

```diff
@@ -20,6 +24,10 @@ class HomeViewModel @Inject constructor(
     val uiModels: StateFlow<List<UiModel>>
         get() = _uiModels
 
+    private val _requestPermissions = MutableStateFlow<Array<out String>>(emptyArray())
+    val requestPermissions: StateFlow<Array<out String>>
+        get() = _requestPermissions
```

Comment:
Shouldn't the view be kept as simple as possible with all logic in the ViewModel? 🤔 
Suppose we have a button that needs to check `LOCATION` permission, so the view's code would look something like
```kotlin
button.setOnClickListener {
    if (hasPermission(LOCATION)) {
         doSomething()
    } else {
        requestPermission(LOCATION)
    }
}
```
instead of just 
```kotlin
button.setOnClickListener { viewModel.doSomething() }
```
Please correct me if I've misunderstood something 😅

Reply:
What if we do this in `HomeFragment.kt?` So no code in `HomeViewmodel.kt` as @luongvo suggested

If the user already has permission, it will go to `Granted`, then we can still call `viewModel.doSomething()` if needed.

```
override fun setupView() {
      button.setOnClickListener { 
         requestPermissions(
               Manifest.permission.ACCESS_FINE_LOCATION,
               Manifest.permission.CAMERA
          ) 
     }
}

private fun requestPermissions(vararg permissions: String) {
    lifecycleScope.launch {
        val result = requestPermissionsAsync(*permissions)
        handlePermissionResult(result)
    }
}

private fun handlePermissionResult(permissionResult: PermissionResult) {
    when (permissionResult) {
        is Granted -> Timber.d("${permissionResult.grantedPermissions} granted")
        is Denied.JustDenied -> Timber.d("${permissionResult.deniedPermissions} denied")
        is Denied.NeedsRationale -> Timber.d("${permissionResult.deniedPermissions} needs rationale")
        is Denied.DeniedPermanently -> Timber.d("${permissionResult.deniedPermissions} denied for good")
        is Cancelled -> Timber.d("request cancelled")
    }
}

```